### DISCUSSION TITLE
KEPServerEX 6 with Python

### DISCUSSION THREAD

**Discussion by AndresParra11** [FreeOpcUa/opcua-asyncio#1350] [Q&A]
In KEPServerEX 6 I have the following configuration (OPC UA Configuration Manager):

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/116961218/0a33c102-aa38-421c-94d3-374f5e7a67a5)

In device 1 channel 1, I have two tags to test the connection:

![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/116961218/4ea692fa-38c1-4e8a-89fd-0b26765330dd)

I have Python 3.11 installed and I'm using Visual Studio Code (VSC). I already installed the asyncua library with the "pip install asyncua" command from the VSC terminal. And I have the following code:

```
import asyncio
import logging
from asyncua import Client, ua

logging.basicConfig(level=logging.WARNING)
_logger = logging.getLogger('asyncua')


async def main():
    client = Client(url="opc.tcp://127.0.0.1:49320", timeout=4)

    # catching all networkrequests is some times important
    try:
        await client.connect()
    except Exception as e:  # catching all exceptions is not always wise!
        print(e)
        return

    '''
    Clientcode!
    '''

    obj = client.get_objects_node()

    children = await obj.get_children()

    for child in children:
        print("Objects-Node-Child: ", child)

    await asyncio.sleep(2)

    try:
        await client.disconnect()
    except Exception as e:  # catching all exceptions is not always wise!
        print(e)
        return

if __name__ == "__main__":
    asyncio.run(main())
```

And it throws me the following errors:

ARNING:asyncua.client.client:Requested session timeout to be 3600000ms, got 60000ms instead
WARNING:asyncua.client.ua_client.UASocketProtocol:ServiceFault (BadIdentityTokenInvalid, diagnostics: DiagnosticInfo(SymbolicId=None, NamespaceURI=None, Locale=None, LocalizedText=None, AdditionalInfo=None, InnerStatusCode=None, InnerDiagnosticInfo=None)) from server received  in response to ActivateSessionRequest
"The session cannot be used because ActivateSession has not been called."(BadSessionNotActivated)

I don't know what to do, if you can help me I appreciate it. Thank you for your time.

**Comment by schroeder-** [comment]
Security None just the secruity of the communication. You maybe still need a useser + password or a certificate, to login. That is what the errorcode BadIdentityTokenInvalid means.

Did you try to connect via UAExpert?

**Comment by rlabbe82** [comment]
Likely username and password. Projects in Kepware don't allow anonymous user by default.
